### PR TITLE
API: PeriodIndex.values now return array of Period objects

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] split from #13941 (comment)
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

This change itself needs more tests / perf testing, split from the original PR.

### CODE REVIEW

**Comment by jreback** [comment]
this looks fine. pls add a small section for showing the changes in whatsnew (though if this should be merged after #13941 can just add on there (and vice-versa)

**Comment by sinhrks** [comment]
OK, let #13941 be first, because this PR includes some workaround which can be removed once we have ``period`` dtype.

**Comment by jreback** [comment]
@sinhrks pls rebase. This will be a welcome change!

**Comment by jorisvandenbossche** [comment]
Needs a new rebase ;) (for your fixed of the tests)

File: pandas/tseries/period.py [hunk 1]
```diff
@@ -349,6 +349,24 @@ def __contains__(self, key):
                 return False
             return False
 
+    @property
+    def asi8(self):
+        return self._values.view('i8')
+
+    @property
+    def values(self):
```
Comment (jorisvandenbossche) [inline pandas/tseries/period.py]: Can you add docstrings for those?

**Comment by jorisvandenbossche** [comment]
Can you also add a whatsnew notice?

**Comment by sinhrks** [comment]
Added whatsnew, now ready for review.

**Comment by jreback** [comment]
thanks!

File: doc/source/whatsnew/v0.19.0.txt [hunk 2]
```diff
@@ -719,6 +722,21 @@ New Behavior:
...
+
+.. code-block:: ipython
+   In [6]: pi = pd.PeriodIndex(['2011-01', '2011-02'], freq='M')
+   In [7]: pi.values
+   array([492, 493])
+
+.. ipython:: python
+
+   pi = pd.PeriodIndex(['2011-01', '2011-02'], freq='M')
+   pi.values
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.19.0.txt]: @sinhrks is it correct to say that to get the integer values as before (if you for some reason were using it), you cas access it through `.asi8` ?

Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: .asi8 is not really public 

.astype is the single method for converting types

Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: further the codes are a private impl

**Pull request closed without merging**